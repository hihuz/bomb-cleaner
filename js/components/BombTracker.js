import React from 'react';
import { connect } from 'react-redux';

const BombTracker = (props) => (
  <div className="bomb-tracker">
    {props.bombsRemaining}
  </div>
);

const mapStateToProps = (state) => {
  return { bombsRemaining: state.grid.bombs - state.grid.flags };
};

export default connect(mapStateToProps)(BombTracker);
