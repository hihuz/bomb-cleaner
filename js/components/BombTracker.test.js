import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import BombTracker from './BombTracker';

test('Snapshot', () => {
  const component = shallow(<BombTracker />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Should display the bombRemainings passed as props', () => {
  const expected = '10';
  const component = shallow(<BombTracker bombsRemaining={expected} />);
  const actual = component.find('.bomb-tracker').text();
  expect(actual).toEqual(expected);
});
