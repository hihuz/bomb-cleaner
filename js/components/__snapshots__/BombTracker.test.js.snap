exports[`test Snapshot 1`] = `
<div
  className="game-infos-item">
  <div
    className="bomb-tracker">
    <i
      className="icon-bomb" />
  </div>
</div>
`;
